### PR TITLE
v2 New Servers System

### PR DESCRIPTION
**Pull request by JonnyWong16**
Todo:
- Store server config in database
- Multi-server support
  - Need input on the best way to do this without adding "server_id" as an argument to every single `PmsConnect()` and `PlexTV()` call, or looping `for server in servers_list`.
- Plex Cloud
- Streaming Brain
- Better http fetch: https://gist.github.com/Hellowlol/1aa04a1808534aee6b87a906b1d7e53c
- Use json instead of XML (maybe?)

### CODE REVIEW

**Comment by Hellowlol** [comment]
In terms on supporting multiple servers are you planning on having a dropdown where you choose server or that you can see all of them at the same time?

**Comment by JonnyWong16** [comment]
A dropdown would be the easy way out. Preferably all servers at the same time with filters like the history table.

**Comment by JonnyWong16** [comment]
@Hellowlol Want to check my urllib3 pool implementation? Currently works with one url. I assume I can use this to request info from multiple servers.

Exceptions need to be updated. Not sure about my SSL implementation.

**Comment by Hellowlol** [comment]
Yeah, you can use it with several servers.

File: plexpy/http_handler.py [hunk 1]
```diff
@@ -30,94 +33,144 @@ class HTTPHandler(object):
...
-        valid_request_types = ['GET', 'POST', 'PUT', 'DELETE']
+            if no_token and headers:
+                self.headers = headers
+            elif headers:
+                self.headers.update(headers)
 
-        if request_type.upper() not in valid_request_types:
-            logger.debug(u"HTTP request made but unsupported request type given.")
+            responses = []
+            for r in self._http_requests_pool(request_urls):
```
[the remark this answers is not in the document]
Comment (JonnyWong16) [reply plexpy/http_handler.py:89]: This is only temporary to make sure it works. I'm only working with one URL at a time at the moment.

**Comment by duramato** [comment]
Gave this a try, 
it wasn't able to retrieve my token automatically i had to manually input it.
```
2017-01-29 18:17:53 - INFO    :: CP Server Thread-14 : PlexPy PlexTV :: Fetching a new Plex.tv token for PlexPy.
2017-01-29 18:17:53 - INFO    :: CP Server Thread-6 : PlexPy PlexTV :: Fetching a new Plex.tv token for PlexPy.
2017-01-29 18:17:53 - INFO    :: CP Server Thread-12 : PlexPy PlexTV :: Fetching a new Plex.tv token for PlexPy.
2017-01-29 18:17:53 - INFO    :: CP Server Thread-5 : PlexPy PlexTV :: Fetching a new Plex.tv token for PlexPy.
2017-01-29 18:17:53 - WARNING :: CP Server Thread-14 : Failed to access uri endpoint /users/sign_in.xml. Is your server maybe accepting SSL connections only? __str__ returned non-string (type Error)
2017-01-29 18:17:53 - WARNING :: CP Server Thread-6 : Failed to access uri endpoint /users/sign_in.xml. Is your server maybe accepting SSL connections only? __str__ returned non-string (type Error)
2017-01-29 18:17:53 - WARNING :: CP Server Thread-12 : Failed to access uri endpoint /users/sign_in.xml. Is your server maybe accepting SSL connections only? __str__ returned non-string (type Error)
2017-01-29 18:17:53 - WARNING :: CP Server Thread-5 : Failed to access uri endpoint /users/sign_in.xml. Is your server maybe accepting SSL connections only? __str__ returned non-string (type Error)
```
And when i open that new server config i get: 
```
2017-01-29 18:10:20 - WARNING :: CP Server Thread-5 : PlexPy PlexTV :: Unable to parse XML for get_plexpass_status: 'NoneType' object has no attribute 'getElementsByTagName'.
2017-01-29 18:10:20 - WARNING :: CP Server Thread-5 : PlexPy PlexTV :: Failed to get servers from plex: 'NoneType' object has no attribute 'getElementsByTagName'.
```
and now info from the recently added server shows on the Home page

It also shows the version as unkown:
img

**Comment by JonnyWong16** [comment]
Why are you using this branch? This branch does not work _**at all**_.

**Comment by duramato** [comment]
I was expecting it would be in a testable manner, as multiple plex media server support is what i need to be able to fully use plexpy.

**Comment by PepsiMaxIsti** [comment]
How can I get the version 2, I now have version 1.4.25

**Comment by PepsiMaxIsti** [comment]
How can I get version 2, i now have 1.4.25, and would like to update to use the app for this.

**Pull request closed without merging**